docs/source/api/notebooks/Notebook_5_GP_solution.ipynb

```diff
@@ -0,0 +1,631 @@
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import torch as th\n",
+    "from graphstorm.model import (GSgnnNodeModel,\n",
+    "                              GSNodeEncoderInputLayer,\n",
+    "                              RelationalGCNEncoder,\n",
+    "                              RelationalGATEncoder,\n",
+    "                              HGTEncoder,\n",
+    "                              EntityClassifier,\n",
+    "                              ClassifyLossFunc)\n",
+    "from graphstorm.model.gnn_encoder_base import GraphConvEncoder\n",
+    "\n",
+    "class GPEncoder4SupernodeOgbg(GraphConvEncoder):\n",
+    "    r\"\"\"A graph conv encoder for Graph Classification\n",
+    "\n",
+    "        Unique arguments in this class:\n",
+    "        -------------------------------\n",
+    "        base_encoder: GraphStorm ConvEncoder types, options:\n",
+    "            1. `RelationalGCNEncoder`;\n",
+    "            2. `RelationalGATEncoder`;\n",
+    "            3. `HGTEncoder`.\n",
+    "        read_out_opt: string in the following options:\n",
+    "            The aggregation method for the cached supernodes' representations. The current options:\n",
+    "            1. `last_only`: only use the last layer's representations. If use this option, \n",
+    "               the read_out_ops will be ignored because there is only one layer representation\n",
+    "               is involved in the final read_out.\n",
+    "            2. `mean`: compute the mean of all of the cached supernode representations.\n",
+    "            3. `sum`: compute the summantion of all of the cached supernode representations.\n",
+    "            4. `weighted_sum`: use additional weight parameters to compute the weighted summation\n",
+    "               all of the cached supernode representations.\n",
+    "            5. `min`: compute the minimum in each dimension of the all of the cached supernode\n",
+    "               representations.\n",
+    "            6. `max`: compute the maximum in each dimension of the all of the cached supernode\n",
+    "               representations.\n",
+    "        super_ntype: string\n",
+    "            The name of supernode type. Default is 'super'.\n",
+    "\n",
+    "    \"\"\"\n",
+    "    def __init__(self,\n",
+    "                 h_dim,\n",
+    "                 out_dim,\n",
+    "                 base_encoder,\n",
+    "                 read_out_opt='last_only',\n",
+    "                 super_ntype='super'\n",
+    "                ):\n",
+    "        assert isinstance(base_encoder, (RelationalGCNEncoder, RelationalGATEncoder, HGTEncoder)), \\\n",
+    "               'Only support GraphStorm\\'s RelationalGCNEncoder, RelationalGATEncoder, and HGTEncoder'\n",
+    "        assert base_encoder.num_layers >= 3, 'For Graph Prediction task, at least two layers GNN' + \\\n",
+    "                                       f'encoder required, but got {num_hidden_layers} ...'\n",
```

Comment:
The assertion text "at least 2 layers" i.e. >=2 differs from the code, >=3

Reply:
Because the `base_encoder.num_layers` is actually the `self._num_hid_layers + 1`, the output will be >=3. The `num_hidden_layers` is an error. Updated to `{base_encoder.num_layers - 1}`.